### PR TITLE
[Tag] Avoid Tag's remove button to shrink (#4815)

### LINKED ISSUE #4815: [Tag] Remove Button is smaller in <Tag> with ellipsis
## Issue summary

When using the `<Tag>` component that:
- Is **dismissible**
- The **content is too long** and gets shortened

**The `Remove` button shrinks**

## Expected behavior

The `Remove` button should stay fixed in size and the space should be taken away from the content instead

## Actual behavior

The `Remove` button shrinks, or rather, it gets pushed slightly to the edge

Screenshot 2021-12-09 at 21 22 55

## Steps to reproduce the problem

Check this [Reduced test case](https://codesandbox.io/s/pensive-payne-7xpmk)

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: latest?
- Browser: Chrome, Safari
- Device: Desktop
- Operating System: OSX

### PR DESCRIPTION
**Pull request by jbalsas**
### WHY are these changes introduced?

Fixes #4815 which causes the `Remove` button within tags with long texts to "shrink"

Screenshot 2021-12-09 at 21 22 55

### WHAT is this pull request doing?

- I think (or that was my intention) that this just adds `flex-shrink: 0` to the button so it won't give way for the text and retain its size.
- I've also added an example in the `Readme` to see this in Storybook, but maybe we don't want to see this in there 🤷 .  I find the ellipsis behaviour interesting to showcase nevertheless.

### How to 🎩

Check the `Tag` section in Storybook

Screenshot 2021-12-09 at 21 48 56

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers): **Chrome, Firefox, Safari**
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] For visual design changes, ping @ sarahill to update the Polaris UI kit

Possible visual changes, so cc'ing @sarahill

### CODE REVIEW

**Comment by ghost** [comment]
👋 Thanks for opening your first pull request. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md).

**Review by kyledurand — COMMENTED** [review]
Nice! My only nit pick would be to group this with the display property. That and please add an entry to UNRELEASED.md

**Comment by jbalsas** [comment]
Hey @kyledurand thanks for the comment! Since this is such a small change I've simply amended the commit and pushforced it here, hope that's okay!

> Nice! My only nit pick would be to group this with the display property. 

Is there somewhere in the contributing guidelines or elsewhere where the ordering rules are stated? The only thing I found was something like: `For code style best practices, take a look at existing components`.  I tried to discern the pattern and it wasn't obvious to me. Could you explain it to me or guide me for future reference?

**Review by dleroux — APPROVED** [review]
Thanks for doing this @jbalsas ! One small suggestion on the release note. Otherwise 🐑 🇮🇹

File: UNRELEASED.md
Comment (dleroux) [inline UNRELEASED.md]:
```suggestion
- Remove Button is smaller in `Tag` with ellipsis ([#4815](https://github.com/Shopify/polaris-react/issues/4815))
```

Comment (jbalsas) [reply UNRELEASED.md]: Fixed and pushed! (again, just amending the commit to avoid adding too much noise to the history!)

**Comment by jbalsas** [comment]
@kyledurand, @dleroux, I think this one is good to go, but there's one action missing in the `UI Tests`

Screenshot 2021-12-10 at 13 13 33

The updates to the `Tag` storybook need baseline approval... what's the workflow for that? should I go and do that or is someone else going to take care of it before merging this in?

**Comment by jbalsas** [comment]
FYI, I'm removing @alex-page as a reviewer since I from what I understood he should be on vacation by now!!

vacation

(I haven't met him, but I guess this look accurate 👆 )

File: UNRELEASED.md
Comment (kyledurand) [inline UNRELEASED.md]:
```suggestion
- Fixed a bug where remove button could shrink in the `Tag` component ([#4816](https://github.com/Shopify/polaris-react/issues/4816))
```

**Review by kyledurand — APPROVED** [review]
There's probably no discernible pattern anymore with so many contributors to the project. It's just a personal preference of mine where I find it easier to parse css with groupings like this. 

We know that Tag > Button is in a flex context and we can easily see that below that is block context 🤷 . This is why I said it was a nit pick 😄 

I just made a suggestion for the unreleased entry. We link to the PR and not the issue

File: UNRELEASED.md
Comment (jbalsas) [reply UNRELEASED.md]: got it!

**Comment by jbalsas** [comment]
Hey @kyledurand!

> There's probably no discernible pattern anymore with so many contributors to the project. It's just a personal preference of mine where I find it easier to parse css with groupings like this.
> 
> We know that Tag > Button is in a flex context and we can easily see that below that is block context 🤷 . This is why I said it was a nit pick 😄

Understood! I'd venture that just sorting alphabetically can make reasoning with the pattern trivial (no reasoning needed) and can be easily automated. That's obviously part of a larger conversation and of course can have a temporary impact on how you scan and make mental models of the code so let's table that for now 😁 

(Or if you wanted to have a pattern... maybe we could look into codifying that in an eslint/stylelint/whateverlint rule?)

> I just made a suggestion for the unreleased entry. We link to the PR and not the issue

Pushed it directly from GitHub.

yolo

**Comment by kyledurand** [comment]
> (Or if you wanted to have a pattern... maybe we could look into codifying that in an eslint/stylelint/whateverlint rule?)

We had that going on in web when it first started up but was more pain than it was worth for a company of our size, unfortunately.

Anyway thanks for contributing 🙌

**Comment by jbalsas** [comment]
> We had that going on in web when it first started up but was more pain than it was worth for a company of our size, unfortunately.

Yeah, that's why I think just sorting alphabetically everywhere could help, because that's easily enforceable and has a really low cognitive load. But I'm not here to sell you on this now 😂 

> Anyway thanks for contributing 🙌

Happy to, looking forward to the next one! 💪 

 Thanks for your time and patience! Let me know if there's anything you'd need from me!

**Comment by jbalsas** [comment]
Saw green and pushed da button, hope that was the right thing to do!!

shipit

**Pull request merged**
